File: examples/sample.php
<?php
// PHP sample

declare(strict_types=1);

enum Tier: string {
    case Free = 'free';
    case Pro = 'pro';
}

final class Plan {
    public function __construct(
        public readonly string $name,
        public readonly Tier $tier,
        public readonly float $price,
    ) {}
}

$plans = [new Plan('Starter', Tier::Free, 0.0), new Plan('Team', Tier::Pro, 29.0)];
$total = array_reduce($plans, fn(float $sum, Plan $p) => $sum + $p->price, 0.0);
echo "Total: {$total}\n";

File: examples/sample.py
# Python sample
from dataclasses import dataclass
from typing import Final

PI: Final[float] = 3.14159

@dataclass
class Circle:
    radius: float

    def area(self) -> float:
        return PI * self.radius ** 2


def summarize(values: list[int]) -> dict[str, float]:
    total = sum(values)
    avg = total / len(values) if values else 0.0
    return {"total": float(total), "avg": avg}


if __name__ == "__main__":
    c = Circle(radius=2.5)
    print(f"area={c.area():.2f}", summarize([1, 2, 3, 4]))
